pkg/gardenlet/apis/config/helper/helpers.go

```diff
@@ -36,6 +38,30 @@ func SeedNameFromSeedConfig(seedConfig *config.SeedConfig) string {
 	return seedConfig.SeedTemplate.Name
 }
 
+// SeedNames returns all seed names matching the given config or LabelSelector if no config is given.
+func SeedNames(seedConfig *config.SeedConfig, seedLister gardencorelisters.SeedLister, labelSelector *metav1.LabelSelector) []string {
```

Comment:
I don't think we should have a function in the `helper` package that itself depends on the lister. Maybe you can change the signature to already provide the `[]*gardencorev1beta1.Seed`?

Reply:
> Maybe you can change the signature to already provide the []*gardencorev1beta1.Seed?

That was actually the motivation to create a new function in the first place, because if you already provided a list with seeds, then the function itself is kind of useless. I can move the whole logic to the [factory](https://github.com/gardener/gardener/blob/f95a15cbe53ea68492fa4da20998828695122bcb/pkg/gardenlet/controller/factory.go#L94) and remove this function or move it elsewhere (e.g. private function in factory)?

Reply:
Sure, that's also fine. I'm only worried about the Go package dependency graph, so moving this to somewhere else is totally fine.